src/web/services/StudentProfile.service.ts

```diff
@@ -0,0 +1,35 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { StudentDetails } from '../app/pages-instructor/student-profile/student-profile';
+import { MessageOutput } from '../types/api-output';
+import { StudentProfileUpdateRequest } from '../types/api-request';
+import { HttpRequestService } from './http-request.service';
+
+/**
+ * Handles student profile related logic provision.
+ */
+
+@Injectable({
+  providedIn: 'root',
+})
+export class StudentProfileService {
+
+  constructor(private httpRequestService: HttpRequestService) {
+  }
+
+  /**
+   * Updates a student profile by calling API.
+   */
+  updateStudentProfile(paramsMap: {[key: string]: string}, requestBody: StudentProfileUpdateRequest)
```

Comment:
Don't rely on the caller to properly populate the params map. You should instead take in the requested values (shortname, email, institute etc.) and then create the params map in this method.

Reply:
well is that the right way to do? Feels like it defeats the purpose of request DTO ?

Reply:
@ChooJeremy Is correct. @junming403 This the actually the purpose of request DTO.